cli/azd/internal/cmd/add/add.go

```diff
@@ -174,6 +204,21 @@ func (a *AddAction) Run(ctx context.Context) (*actions.ActionResult, error) {
 		if err != nil {
 			return nil, fmt.Errorf("appending resource: %w", err)
 		}
+
+		// Add dependent resources to 'uses'
+		for _, dep := range dependentResources {
+			// Skip if already in 'uses'
+			if slices.Contains(prjConfig.Resources[svc].Uses, dep.Name) {
+				continue
+			}
+			err = yamlnode.Append(&doc, fmt.Sprintf("resources.%s.uses[]?", svc), &yaml.Node{
```

Comment:
I feel that a discussion is warranted on whether `uses` would mean the keyvault relationship here. Perhaps we should restrict `uses` in the schema for now to just "host" applications to mean "connects to resources"

Reply:
I think it would be better to avoid automatically adding `uses`.. For vault in particular, let's just keep it as an implicit "uses".

Generally speaking, the dependency relationship ought not to always be structured that way, so I have a slight preference here to remove this logic.

Reply:
Updated, thanks. Also realized with the current `resources.bicept` this isn't really doing anything either.